enos/ansible/roles/bench/tasks/rally.yml

```diff
@@ -52,35 +52,37 @@
   copy: src={{ bench.plugin_location }} dest=/root/rally_home/.rally/plugins owner=655500
   when: bench.plugin_location is defined
 
+- name: create arg file
+  copy:
+    content: "{{ bench.args }}"
+    dest: /root/rally_home/rally-args.json
 
-- name: debug
-  debug: msg="rally task start {{ bench.scenario_location | basename }} --task-args '{{ bench.args }}'"
-
-# NOTE(msimonin): ansible 2.2.0 breaks the return value.
-# see https://github.com/BeyondTheClouds/enos/issues/41
-- name: Run rally benchmark
+- name: Run benchmark
   docker_container:
     name: "{{ bench.scenario_location | to_uuid }}"
     image: rallyforge/rally
     state: started
     volumes:
       - /root/rally_home:/home/rally
-    command: rally task start {{ bench.scenario_location | basename }} --task-args '{{ bench.args }}'
+    command: rally task start {{ bench.scenario_location | basename }} --task-args-file /home/rally/rally-args.json
   register: docker_output
 
 - name: Wait for the end of the test, this may take a while...
-  command: "docker ps -q --filter id={{ docker_output.ansible_facts.ansible_docker_container.Id }}"
+  command: "docker ps -q --filter id={{ docker_output.ansible_facts.docker_container.Id }}"
   register: finished
   until: finished.stdout == ""
   delay: 20
   retries: 10000
 
-- name: List available rally reports
-  command: docker run -v /root/rally_home:/home/rally rallyforge/rally  rally task list --uuids-only
-  register: list
+- name: List available last available report
```

Comment:
List ~~available~~ last available report. But, I would say: "Find the last available report" because this is no more a list.